lib/onboarding.py

```diff
@@ -1,20 +1,15 @@
 import requests
 import sublime
 
-from ..lib import link_opener
+from ..lib import link_opener, languages, settings
 
-def start_onboarding():
-    """Attempts to open the live onboarding file. If fetching the file fails,
-    then a help dialog is shown instead.
-
-    This function returns whether or not the help prompt should be suppressed
-    in future startups. The help prompt should be suppressed if fetching the
-    live onboarding file succeeds or if the user clicked on the "No" button
-    in the help dialog.
+def start_onboarding(ext):
```

Comment:
I would move all the first-time onboarding logic here and follow the pattern Python uses:

1. Check `kited` to see if there's an onboarding file for this language. If there is, show it and also prevent the onboarding file from being shown again by disabling the setting.
2. If there isn't an onboarding file, show the help dialog. For JS and Go, that's the beta notification. If the user chooses to hide this dialog by clicking "Hide Forever", prevent the onboarding flow for this language from taking place by disabling the setting as in step 1.

This means that beta JS users may never see the onboarding file when there is one, which I think is fine. By then, the beta users would already be familiar with the product (it's just completions) and seeing another onboarding file would probably be an unexpected experience. I also don't think we should add too much additional complexity just for a beta notification.

Reply:
Ah I mixed that up in my head with the per language notifications. The spec states showing only one onboarding file clearly. Will change!

Reply:
Hmm, I figure won't always be in beta for JS and go but we might have some future beta languages, so the fallback message should be generic and not specific to the beta. What are your thoughts on going with a single `show_<lang>_onboarding` that shows one of:

1. onboarding file (if available)
2. beta notification (if beta)
3. generic message (final fallback)

Reply:
I think even after the beta, we'll need a language specific fallback dialog. The current fallback dialog opens up a [help article](https://help.kite.com/article/76-using-the-sublime-text-plugin) focused on Python features. You can imagine that we'll eventually have help articles for Go and JS.

#2 and #3 above should be the same thing.

1. Show onboarding file if available
2. Show help dialog if not

For this PR, the help dialogues for JS and Go have beta copy. Once we're out of beta, we'll update it with a link to the language specific help article.

Reply:
A higher level comment: I think it's fine to introduce new code paths for beta functionality if it's warranted. For example, if we were testing a new feature, or doing A/B testing, then it makes sense to try to abstract that logic somewhere. But if the only difference between the beta and prod is that the beta notification copy is different, I think it's cleaner to implement this PR as though there was no beta, and just update the copy later.